xbmc/addons/Addon.cpp

```diff
@@ -622,6 +621,52 @@ AddonVersion CAddon::GetDependencyVersion(const std::string &dependencyID) const
   return AddonVersion("0.0.0");
 }
 
+
+void OnPreInstall(const AddonPtr& metaAddon)
```

Comment:
I double checked, and at least for `OnPreInstall` and `OnPostInstall`, I think is it (see `CAddonInstaller::Install`). When will you ever install one you already have? The point I'm trying to make is that it may not be the real one and you should always assume it isn't.